packages/pnpm/src/main.ts

```diff
@@ -170,9 +170,11 @@ export default async function run (inputArgv: string[]) {
     config.filter = config.filter ?? []
     config.filterProd = config.filterProd ?? []
 
+    const extendFilterPattern = (filter: string) => (filter.match(/^[a-z]/) != null) ? `@*/${filter}` : filter
```

Comment:
just in case:
- `/regexp/.test(filter)` might be significantly [faster](https://stackoverflow.com/a/10940138)
- I checked our Nexus npm registry - there are a lot of packages with name and/or scope which starts from digit, or from capital letter. Package name examples: `1.1.1`, `@A.a-a/42a`, `@aaa/A.a-a13`, `@aaa/42b`.
`pnpm view <PACKAGE> versions` works with these packages just fine.